src/cfile/writer.py

```diff
@@ -599,19 +611,60 @@ def _write_func_call(self, elem: core.FunctionCall) -> None:
             self._write_expression(arg)
         self._write(")")
 
+    def _write_enum_usage(self, elem: core.Enum) -> None:
+        """
+        Writes enum usage
+        """
+        if elem.name == "":
+            raise ValueError("enum doesn't have a name. Did you mean to use a declaration?")
+        self._write(f"enum {elem.name}")
+
+    def _write_enum_declaration(self, elem: core.Enum):
+        """
+        Writes enum declaration
+        """
+        self._write(f"enum{" ".join([' __attribute__(('+attribute+'))' for attribute in elem.attributes])} {elem.name}")
```

Comment:
This change only works for GCC-like compilers. Needs some more consideration and design (for example how to specify compiler family).

Reply:
Agree